airbyte_cdk/sources/declarative/manifest_declarative_source.py

```diff
@@ -255,6 +289,9 @@ def connection_checker(self) -> ConnectionChecker:
             )
 
     def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        if self._spec_component:
+            self._spec_component.validate_config(config)
```

Comment:
i feel like i want to document somewhere that the `config` that arrives in every connector came from reading the same config that was rewritten earlier in `__init__`.

And I think just as a whole in our ideal world, we should not even take in the `config` from the method at all since the official source of truth is the `ManifestDeclarativeSource._config` value.

But I digress. No action here because I don't think this PR should concern itself with the removal of incoming `config` from read/check/discover. But in theory after this change, it should be unneeded

One day we'll eventually get to this!
https://github.com/airbytehq/airbyte-internal-issues/issues/13101